Only use C++20 when enabling BLAKE3_USE_TBB

oneTBB parallelism is noted to require a C++20 capable
compiler. Thus, the standard should not be enforced unless
BLAKE3_USE_TBB is being used.

See: https://github.com/BLAKE3-team/BLAKE3/tree/1.8.7/c#cmake
===================================================================
--- c/CMakeLists.txt.orig	2026-08-20 04:50:34.000000000 -0400
+++ c/CMakeLists.txt	2026-09-03 21:50:24.000000000 -0400
@@ -155,8 +155,10 @@
 )
 target_compile_features(blake3 PUBLIC c_std_99)
 if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.12)
-  target_compile_features(blake3 PUBLIC cxx_std_20)
-  # else: add it further below through `BLAKE3_CMAKE_CXXFLAGS_*`
+  if(BLAKE3_USE_TBB)
+    target_compile_features(blake3 PUBLIC cxx_std_20)
+    # else: add it further below through `BLAKE3_CMAKE_CXXFLAGS_*`
+  endif()
 endif()

 # ensure C_EXTENSIONS OFF is respected without overriding CMAKE_C_STANDARD
